code/game/machinery/computer/atmos_alert.dm

```diff
@@ -56,9 +61,20 @@
 	priority_alarms.Cut()
 	minor_alarms.Cut()
 
+	// An area list used for station_only circuits, so we only send an alarm if we're in one of the station or mining home areas
+	var/list/station_alert_areas
+	station_alert_areas = GLOB.the_station_areas + typesof(/area/mine)
+	// Setting up a variable for checking our circuit's station_only
+	var/obj/item/circuitboard/computer/atmos_alert/my_circuit = circuit
 	for (var/obj/machinery/airalarm/air_alarm as anything in GLOB.air_alarms)
-		if (air_alarm.z != z)
-			continue
+		if(!my_circuit.station_only)
+			// By default it only checks if alarms match the console's z-level
+			if (air_alarm.z != z)
+				continue
+		else
+			// If the circuit has station_only, instead checks if alarm areas are in the station list
+			if (!(air_alarm.my_area.type in station_alert_areas))
+				continue
```

Comment:
```suggestion
		// If the circuit has station_only, check if alarm areas are in the station list
		if(my_circuit.station_only)
			if (!(air_alarm.my_area.type in station_alert_areas))
				continue
		// Otherwise just check if alarms match the console's z-level
		else if (air_alarm.z != z)
			continue
```
don't ! a var at the start of an if chain if you can help it, it's essentially a double negative and it can be confusing